graphql/mutation_resolver.go

```diff
@@ -149,18 +149,29 @@ func (r *mutationResolver) SetAnnotationMetadataLinks(ctx context.Context, taskI
 
 // SaveAdminSettings is the resolver for the saveAdminSettings field.
 func (r *mutationResolver) SaveAdminSettings(ctx context.Context, adminSettings restModel.APIAdminSettings) (*restModel.APIAdminSettings, error) {
-	oldSettings, err := evergreen.GetConfigWithoutSecrets(ctx)
+	oldSettings, err := evergreen.GetConfig(ctx)
 	if err != nil {
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting Evergreen configuration: %s", err.Error()))
 	}
-	_, err = data.SetEvergreenSettings(ctx, &adminSettings, oldSettings, mustHaveUser(ctx), true)
+	newSettings, err := data.SetEvergreenSettings(ctx, &adminSettings, oldSettings, mustHaveUser(ctx), false)
 	if err != nil {
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("setting admin settings: %s", err.Error()))
 	}
+	if err = newSettings.Validate(); err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("setting admin settings: %s", err.Error()))
+	}
 
-	newSettings, err := evergreen.GetConfig(ctx)
```

Comment:
you need to keep this in if you want the tests to pass since the "newSettings" that's returned from the save function is redacted

Reply:
Although, I don't think its necessary to return a fully unredacted settings struct on this resolver. I personally think we should return the redacted version since this resolver doesn't need to return secrets. But, I think you'd know best what to return since you use this resolver on the front end! 


If you do decide to return a redacted version of the settings, I'd still like to test calling this to make sure we are getting and setting on AWS correctly.